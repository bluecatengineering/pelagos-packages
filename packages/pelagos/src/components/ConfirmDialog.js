import PropTypes from 'prop-types';
import {t} from '@bluecateng/l10n.macro';
import WarningAltFilled from '@carbon/icons-react/es/WarningAltFilled';

import Dialog from './Dialog';
import Button from './Button';
import './ConfirmDialog.less';

/** A confirmation dialog. */
const ConfirmDialog = ({title, body, confirmText, size, onClose, onConfirm}) => (
	<Dialog title={title} role="alertdialog" size={size}>
		<div className="ConfirmDialog__body">
			<WarningAltFilled className="ConfirmDialog__icon" size={20} />
			<p className="ConfirmDialog__text">{body}</p>
		</div>
		<div>
			<Button id="closeDialogBtn" text={t`Cancel`} onClick={onClose} />
			<Button id="confirmBtn" text={confirmText} type="primary" onClick={onConfirm} />
		</div>
	</Dialog>
);

ConfirmDialog.propTypes = {
	/** The dialog title. */
	title: PropTypes.string,
	/** The text to display in the dialog body. */
	body: PropTypes.string,
	/** The text to display in the confirm button. */
	confirmText: PropTypes.string,
	/** The dialog size. */
	size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
	/** Function invoked when the close button is clicked. */
	onClose: PropTypes.func,
	/** Function invoked when the confirm button is clicked. */
	onConfirm: PropTypes.func,
};

ConfirmDialog.defaultProps = {
	size: 'sm',
};

export default ConfirmDialog;
