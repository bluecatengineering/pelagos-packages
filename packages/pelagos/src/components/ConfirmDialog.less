@import '../../less/spacing';

.ConfirmDialog__body {
	flex-direction: row;
}

.ConfirmDialog__icon {
	flex: none;
	align-self: flex-start;
	color: var(--support-warning);
	margin-right: @sp-16;
	margin-top: @sp-04;
	font-size: 20px;
}

.ConfirmDialog__text {
	overflow-x: hidden;
	overflow-y: auto;
}
