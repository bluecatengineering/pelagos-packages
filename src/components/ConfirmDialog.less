@import '../../less/spacing';

.ConfirmDialog__body {
	flex-direction: row;
	align-items: flex-start;
	max-width: 30em;
}

.ConfirmDialog__icon {
	flex: none;
	color: var(--warn-color);
	margin-right: @sp-08;
	margin-top: @sp-04;
	font-size: 19px;
}
