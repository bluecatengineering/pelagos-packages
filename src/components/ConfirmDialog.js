import PropTypes from 'prop-types';
import {t} from '@bluecat/l10n.macro';
import {faTriangleExclamation} from '@fortawesome/free-solid-svg-icons';

import Dialog from './Dialog';
import SvgIcon from './SvgIcon';
import Button from './Button';
import './ConfirmDialog.less';

const ConfirmDialog = ({title, body, confirmText, size, onClose, onConfirm}) => (
	<Dialog title={title} role="alertdialog" size={size}>
		<div className="ConfirmDialog__body">
			<SvgIcon className="ConfirmDialog__icon" icon={faTriangleExclamation} />
			<p className="ConfirmDialog__text">{body}</p>
		</div>
		<div>
			<Button id="closeDialogBtn" text={t`Cancel`} onClick={onClose} />
			<Button id="confirmBtn" text={confirmText} type="primary" onClick={onConfirm} />
		</div>
	</Dialog>
);

ConfirmDialog.propTypes = {
	/** The dialog title. */
	title: PropTypes.string,
	/** The text to display in the dialog body. */
	body: PropTypes.string,
	/** The text to display in the confirm button. */
	confirmText: PropTypes.string,
	/** The dialog size. */
	size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg']),
	/** Function invoked when the close button is clicked. */
	onClose: PropTypes.func,
	/** Function invoked when the confirm button is clicked. */
	onConfirm: PropTypes.func,
};

ConfirmDialog.defaultProps = {
	size: 'sm',
};

export default ConfirmDialog;
