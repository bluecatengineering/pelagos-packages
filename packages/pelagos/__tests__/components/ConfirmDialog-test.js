import {shallow} from 'enzyme';

import ConfirmDialog from '../../src/components/ConfirmDialog';

jest.unmock('../../src/components/ConfirmDialog');

describe('ConfirmDialog', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<ConfirmDialog className="TestClass" title="Title" body="Body" confirmText="Confirm" />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('behaviour', () => {
		it('calls onClose when the close button is clicked', () => {
			const onClose = jest.fn();
			const wrapper = shallow(<ConfirmDialog onClose={onClose} />);
			wrapper.find('#closeDialogBtn').prop('onClick')();
			expect(onClose).toHaveBeenCalledTimes(1);
		});

		it('calls onConfirm when the close button is clicked', () => {
			const onConfirm = jest.fn();
			const wrapper = shallow(<ConfirmDialog onConfirm={onConfirm} />);
			wrapper.find('#confirmBtn').prop('onClick')();
			expect(onConfirm).toHaveBeenCalledTimes(1);
		});
	});
});
