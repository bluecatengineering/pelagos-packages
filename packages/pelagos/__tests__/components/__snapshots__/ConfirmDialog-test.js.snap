// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog rendering renders expected elements 1`] = `
<Dialog
  role="alertdialog"
  size="sm"
  title="Title"
>
  <div
    className="ConfirmDialog__body"
  >
    <WarningAltFilled
      className="ConfirmDialog__icon"
      size={20}
    />
    <p
      className="ConfirmDialog__text"
    >
      Body
    </p>
  </div>
  <div>
    <Button
      id="closeDialogBtn"
      text="Cancel"
    />
    <Button
      id="confirmBtn"
      text="Confirm"
      type="primary"
    />
  </div>
</Dialog>
`;
